packages/core/src/tracing/instrumentation/protocols/httpClient.js

```diff
@@ -336,16 +336,27 @@ function constructFromUrlOpts(options, self, forceHttps) {
 
   try {
     const agent = options.agent || self.agent;
-    const port = options.port || options.defaultPort || (agent && agent.defaultPort) || 80;
+    const isHttps = forceHttps || options.protocol === 'https:' || (agent && agent.protocol === 'https:');
```

Comment:
```suggestion
    const isHttps = forceHttps || options.protocol === 'https:' || agent?.protocol === 'https:';
```

We should use optional chaining as it is the new syntax!